### PR TITLE
Test proxy start

### PR DESCRIPTION
**Pull request by gearama**
Startign test proxy based on ENV : AZURE_TEST_USE_TEST_PROXY =  ON 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/attestation/azure-security-attestation/test/ut/CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,8 @@ add_compile_definitions(AZURE_TEST_DATA_PATH="${CMAKE_BINARY_DIR}")
 # Export the test folder for recordings access.
 add_compile_definitions(AZURE_TEST_RECORDING_DIR="${CMAKE_CURRENT_LIST_DIR}")
 
+add_compile_definitions(AZURE_TEST_ASSETS_DIR="${CMAKE_CURRENT_LIST_DIR}/../../../")
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/test/ut/CMakeLists.txt:20]: This look incredibly fragile. Is there another way of expressing the service root that doesn't require relative navigation?

File: sdk/attestation/azure-security-attestation/test/ut/CMakeLists.txt [hunk 2]
```diff
@@ -40,3 +42,10 @@ target_link_libraries(azure-security-attestation-test PRIVATE Azure::azure-secur
 gtest_discover_tests(azure-security-attestation-test 
     DISCOVERY_TIMEOUT 600 
     TEST_PREFIX azure-security-attestation.)
+
+
+file(COPY ${CMAKE_CURRENT_LIST_DIR}/../../../../../eng/scripts/Start-TestProxy.ps1
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/test/ut/CMakeLists.txt:47]: Same as earlier - this relative path looks incredibly fragile. Is it possible to do something relative to the root directory here?

File: sdk/storage/azure-storage-common/test/ut/test_base.cpp [hunk 3]
```diff
@@ -34,14 +34,14 @@ namespace Azure { namespace Storage { namespace Blobs { namespace Models {
...
 
-  constexpr static const char* StandardStorageConnectionStringValue = "";
-  constexpr static const char* PremiumStorageConnectionStringValue = "";
-  constexpr static const char* BlobStorageConnectionStringValue = "";
-  constexpr static const char* PremiumFileConnectionStringValue = "";
-  constexpr static const char* AdlsGen2ConnectionStringValue = "";
-  constexpr static const char* AadTenantIdValue = "";
-  constexpr static const char* AadClientIdValue = "";
-  constexpr static const char* AadClientSecretValue = "";
+  constexpr static const char* StandardStorageConnectionStringValue = "=DefaultEndpointsProtocol=https;AccountName=gearamastr;AccountKey=daHwPhp+1UoOMaW552WVKthfT8lMD2Q7vRSonSp+JwFpYPq5/KHm3DVNbwnIecHoh67AnVTvS2ZF+AStjf0b8w==;EndpointSuffix=core.windows.net";
```
Comment (LarryOsterman) [inline sdk/storage/azure-storage-common/test/ut/test_base.cpp]: PII in the code review?

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp [hunk 4]
```diff
@@ -406,9 +406,34 @@ namespace Azure { namespace Core { namespace Test {
      */
     void TearDown() override;
```
Comment (LarryOsterman) [inline sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:408]: THere are a bunchof whitespace differences here that are confusing the diff, not sure why they are there.

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp [hunk 5]
```diff
@@ -406,9 +406,34 @@ namespace Azure { namespace Core { namespace Test {
...
+      {
+        int result
+            = std::system("pwsh -NoProfile -ExecutionPolicy Unrestricted Stop-TestProxy.ps1");
+        std::cout << "Non zero exit code for stop proxy : " << result;
+      }
+    };
     /**
      * Returns the assets.json file path used when invoking the test-proxy playback/record
      */
-    virtual std::string GetAssetsPath() { return "assets.json"; }
```
Comment (LarryOsterman) [inline sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:412]: Is this even used?

**Review by LarryOsterman — CHANGES_REQUESTED** [review]

File: sdk/attestation/azure-security-attestation/test/ut/CMakeLists.txt (hunk 1 above)
Comment (gearama) [reply sdk/attestation/azure-security-attestation/test/ut/CMakeLists.txt:20]: not that i could find,

File: sdk/storage/azure-storage-common/test/ut/test_base.cpp (hunk 3 above)
Comment (gearama) [reply sdk/storage/azure-storage-common/test/ut/test_base.cpp]: fixed

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp (hunk 5 above)
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:412]: yes, somewhere else

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp (hunk 4 above)
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:408]: i don't see those .

**Comment by Jinming-Hu** [comment]
What is the problem we're trying to solve in this PR anyway? Is it because we want to save a step of starting a powershell window and running test-proxy?


I don't like some parts of the implementation, like the relative paths as @LarryOsterman mentioned, and starting a new process in `SetUpTestSuiteLocal` function. Usually another process needs some time to set up, it being started doesn't mean it's ready to serve requests. We may need to wait for 1 or 2 seconds if we want to make it work on most machines. These are fragile. I don't think it's worth the complexity if we only want to save a step of running a command.

**Comment by gearama** [comment]
@LarryOsterman  you requested changes

File: sdk/attestation/azure-security-attestation/test/ut/tpmattestation_test.cpp [hunk 6]
```diff
@@ -25,6 +25,9 @@ namespace Azure { namespace Security { namespace Attestation { namespace Test {
 
   // cspell: words aikcert
   class TpmAttestationTests : public Azure::Core::Test::TestBase {
+  public:
+    TpmAttestationTests() { TestBase::SetUpTestSuiteLocal(AZURE_TEST_ASSETS_DIR); };
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/test/ut/tpmattestation_test.cpp:29]: Why is this only needed for TPM attestation and not all the other test classes? Is there a problem if TPM attestation isn't set up?

**Review by LarryOsterman — APPROVED** [review]

**Comment by LarryOsterman** [comment]
Looks much better, I'm confused about the TPM attestation specific changes, I'm not sure what happens if the TPM attestation tests are not run - it seems like the function being called is kinda important.

File: sdk/attestation/azure-security-attestation/test/ut/tpmattestation_test.cpp (hunk 6 above)
Comment (gearama) [reply sdk/attestation/azure-security-attestation/test/ut/tpmattestation_test.cpp:29]: i couldn't figure out where the other base classes are in attestation. To enable it we woudl need to make the same changes to attestation

**Pull request merged**